mlrun/datastore/targets.py

```diff
@@ -1467,6 +1472,469 @@ def as_df(
+                                   schema=schema, primary_key_column=key)
+
+        :param db_path:                     url string connection to sql database.
+                                            If not set, the SQL_DB_PATH_STRING environment variable will be used.
+        :param collection_name:             the name of the collection to access,
+                                            from the current database
+        :param schema:                      the schema of the collection (must pass when
+                                            create_collection=True)
+        :param primary_key_column:          the primary key of the collection (must pass always)
+        :param if_exists:                   {'fail', 'replace', 'append'}, default 'fail'
+                                            - fail: If table exists, do nothing.
+                                            - replace: If table exists, drop it, recreate it, and insert data.
+                                            - append: If table exists, insert data. Create if does not exist.
+        :param create_collection:           pass True if you want to create new collection named by
+                                            collection_name with schema on current database.
+        :param create_according_to_data:    (not valid)
+        """
+        db_path = db_path or os.getenv(self._SQL_DB_PATH_STRING_ENV_VAR)
+        if db_path is None or collection_name is None:
+            attr = {}
+        else:
+            # check for collection existence and acts according to the user input
+            self._primary_key_column = primary_key_column
+            import sqlalchemy as db
+
+            engine = db.create_engine(db_path)
+            sql_connection = engine.connect()
+            metadata = db.MetaData()
+            collection_exists = engine.dialect.has_table(
+                sql_connection, collection_name
+            )
+            if not collection_exists and not create_collection:
+                raise ValueError(f"Collection named {collection_name} is not exist")
+
+            elif not collection_exists and create_collection:
+                # creat new collection with the given name
+                columns = []
+                for col, col_type in schema.items():
+                    if col_type == int:
+                        col_type = db.Integer
+                    elif col_type == str:
+                        col_type = db.String
+                    elif col_type == datetime.datetime or col_type == pd.Timestamp:
+                        col_type = db.DateTime
+                    elif col_type == bool:
+                        col_type = db.Boolean
+                    elif col_type == float:
+                        col_type = db.Float
+                    else:
+                        raise TypeError(f"{col_type} unsupported type")
```

Comment:
replace with a map:
```
TYPE_TO_SQL_TYPE = {
    int: db.Integer,
    str: db.String,
    datetime.datetime: db.DateTime,
    pd.Timestamp: db.DateTime,
    bool: db.Boolean,
    float: db.Float,
}
```

you may check `if col_type not in TYPE_TO_SQL_TYPE: raise ...`